sharedbuffers/mapped_struct.py

```diff
@@ -1560,21 +1560,23 @@ def proxied_list_richcmp(a, b, op):
         else:
             raise NotImplementedError
 
-    _cmp = proxied_list_cmp(a, b)
     if op == 2: # Py_EQ
-        return _cmp == 0
+        return proxied_list_eq(a, b)
```

Comment:
What's the difference between this and using `proxied_list_cmp` for everything? Is it that we don't use our custom `dict_cmp` when it's not necessary?

Reply:
The difference is that we don't need a `dict_cmp` at all and can throw the same exception python dicts throw.